src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -317,26 +329,40 @@ export class NotebookModel extends Disposable implements INotebookModel {
 				this.notifyError(localize('changeKernelFailed', 'Failed to change kernel: {0}', notebookUtils.getErrorMessage(err)));
 				// TODO should revert kernels dropdown
 			});
-
 	}
 
-	public changeContext(host: string): void {
+	public changeContext(host: string, newConnection?: IConnectionProfile): void {
 		try {
-			let newConnection: IConnectionProfile = this._activeContexts.otherConnections.find((connection) => connection.options['host'] === host);
+			if (!newConnection) {
+				newConnection = this._activeContexts.otherConnections.find((connection) => connection.options['host'] === host);
+			}
 			if (!newConnection && this._activeContexts.defaultConnection.options['host'] === host) {
 				newConnection = this._activeContexts.defaultConnection;
 			}
-			if (newConnection) {
-				SparkMagicContexts.configureContext(newConnection, this.notebookOptions);
-				this._hadoopConnection = new NotebookConnection(newConnection);
-				this._clientSession.updateConnection(this._hadoopConnection);
-			}
+			SparkMagicContexts.configureContext(this.notebookOptions);
+			this._hadoopConnection = new NotebookConnection(newConnection);
+			this.refreshConnections(newConnection);
+			this._clientSession.updateConnection(this._hadoopConnection);
 		} catch (err) {
 			let msg = notebookUtils.getErrorMessage(err);
 			this.notifyError(localize('changeContextFailed', 'Changing context failed: {0}', msg));
 		}
 	}
 
+	private refreshConnections(newConnection: IConnectionProfile) {
+		if (this.isValidKnoxConnection(newConnection) &&
+                this._hadoopConnection.connectionProfile.id !== '-1' &&
+                this._hadoopConnection.connectionProfile.id !== this._activeContexts.defaultConnection.id) {
```

Comment:
Minor: for multi-line clauses like this we often put the `) {` on its own line so the indentation for the rest of the if clause "looks right"